### PR TITLE
Add legsta

### PR DESCRIPTION
**Pull request by emarinier**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Adds [legsta](https://github.com/tseemann/legsta), a tool for performing in silico sequence based typing of Legionella pneumophila.

### CODE REVIEW

File: tools/legsta/legsta.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	<param type="data" name="contigs" format="genbank,embl,gff,gff3,fasta,fastq,clustal,stockholm" multiple="true" label="Contigs" help="One or more input files in FASTA, FASTQ, Genbank, EMBL, Clustal, Stockholm or GFF format." />
+	<param name="noheader" type="boolean" argument="--noheader" truevalue="--noheader" falsevalue="" label="Remove table header" help="Removes the table header from the output (ie. FILE SBT flaA ...)" />
+	<param name="csv" type="boolean" argument="--csv" truevalue="--csv" falsevalue="" label="CSV output" help="Prints the output in comma-separated value format instead of tab-separated value format." />
+    </inputs>
+
+    <outputs>
+        <data name="output" format="tsv">
+            <change_format>
+                <when input="csv" value="--csv" format="csv" />
+                <when input="csv" value="" format="tsv" />
```
Comment (bgruening) [inline tools/legsta/legsta.xml:32]: tabular is better

File: tools/legsta/legsta.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<tool id="legsta" name="legsta" version="@TOOL_VERSION@" profile="20.05">
+    <description>Legionella pneumophila sequence based typing</description>
+
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+
+    <command detect_errors="exit_code"><![CDATA[
+        legsta
```
Comment (bgruening) [inline tools/legsta/legsta.xml:10]: can you please use everywhere 4 spaces instead of tab?

File: tools/legsta/legsta.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        #for $i in $contigs
+            '$i'
+        #end for
+
+        > '$output'
+    ]]></command>
+
+    <inputs>
+	<param type="data" name="contigs" format="genbank,embl,gff,gff3,fasta,fastq,clustal,stockholm" multiple="true" label="Contigs" help="One or more input files in FASTA, FASTQ, Genbank, EMBL, Clustal, Stockholm or GFF format." />
```
Comment (bgruening) [inline tools/legsta/legsta.xml:23]: please fix the indentation

File: tools/legsta/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: legsta
+owner: iuc
+description: Performs in silico Legionella pneumophila sequence based typing.
+long_description: Provides a TSV output file that reveals the sequence based type of provided Legionella pneumophila inputs.
+categories:
+          - Sequence Analysis
```
Comment (bgruening) [inline tools/legsta/.shed.yml:6]: indentation

File: tools/legsta/legsta.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	<param name="noheader" type="boolean" argument="--noheader" truevalue="--noheader" falsevalue="" label="Remove table header" help="Removes the table header from the output (ie. FILE SBT flaA ...)" />
+	<param name="csv" type="boolean" argument="--csv" truevalue="--csv" falsevalue="" label="CSV output" help="Prints the output in comma-separated value format instead of tab-separated value format." />
+    </inputs>
+
+    <outputs>
+        <data name="output" format="tsv">
```
Comment (bgruening) [inline tools/legsta/legsta.xml:29]: tabular

**Review by bgruening — COMMENTED** [review]
Can you please reduce the size of your test file. It needs to be below 1MB.

Thanks a lot for your contribution @emarinier

File: tools/legsta/legsta.xml (hunk 1 above)
Comment (emarinier) [reply tools/legsta/legsta.xml]: I've changed "tsv" -> "tabular".

File: tools/legsta/legsta.xml (hunk 5 above)
Comment (emarinier) [reply tools/legsta/legsta.xml]: I've changed "tsv" -> "tabular".

File: tools/legsta/legsta.xml (hunk 3 above)
Comment (emarinier) [reply tools/legsta/legsta.xml]: Fixed.

File: tools/legsta/legsta.xml (hunk 2 above)
Comment (emarinier) [reply tools/legsta/legsta.xml:10]: I believe I've changed them all to 4 spaces.

**Comment by emarinier** [comment]
> Can you please reduce the size of your test file. It needs to be below 1MB.
> 
> Thanks a lot for your contribution @emarinier

I've changed the test file to be compressed, which now uses 985K.

File: tools/legsta/.shed.yml (hunk 4 above)
Comment (emarinier) [reply tools/legsta/.shed.yml]: Fixed.

**Review by bgruening — COMMENTED** [review]
@emarinier do you know if this tool can deal with native fasta.gz files? If so we should add support for that I think.

**Comment by emarinier** [comment]
@bgruening 

> @emarinier do you know if this tool can deal with native fasta.gz files? If so we should add support for that I think.

Yes, apparently it supports all of the file formats I included, but also compressed in `.gz`, `.bz2`, and `.zip` formats:

https://github.com/tseemann/any2fasta#motivation
(the above link is relevant documentation from conversion tool used by legsta)

I didn't see the compressed versions explicitly mentioned in the [sample formats file](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/datatypes_conf.xml.sample). Do I just append the compression extensions to each of them?

` format="genbank,genbank.gz,genbank.bz2,genbank.zip,embl,embl.gz,embl.bz2,embl.zip, ...  "`

**Comment by bgruening** [comment]
> Yes, apparently it supports all of the file formats I included, but also compressed in .gz, .bz2, and .zip formats:

did you test this on commandline? Galaxy will unpack those file for you before passing to the tool if you do not make this explicit. So make sure that the tool really receives a fastq.gz file.

**Comment by emarinier** [comment]
> > Yes, apparently it supports all of the file formats I included, but also compressed in .gz, .bz2, and .zip formats:
> 
> did you test this on commandline? Galaxy will unpack those file for you before passing to the tool if you do not make this explicit. So make sure that the tool really receives a fastq.gz file.

**Testing .fna, .fna.bz2, .fna.gz, .fna.zip:**

```
(base) eric@ragnarok:~/Downloads$ legsta NC_018140.*
Found 'any2fasta' => /home/eric/miniconda3/bin/any2fasta
Found 'isPcr' => /home/eric/miniconda3/bin/isPcr
Loaded 2794 SBT types.
Loaded 39 flaA alleles
Loaded 58 pilE alleles
Loaded 77 asd alleles
Loaded 91 mip alleles
Loaded 105 mompS alleles
Loaded 58 proA alleles
Loaded 96 neuA alleles
FILE    SBT     flaA    pilE    asd     mip     mompS   proA    neuA
Running: any2fasta -q -u NC_018140\.fna | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140.fna   734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140\.fna\.bz2 | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140.fna.bz2       734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140\.fna\.gz | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140.fna.gz        734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140\.fna\.zip | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
```

**Testing .gbk, .gbk.bz2, .gbk.gz, .gbk.zip:**

```
(base) eric@ragnarok:~/Downloads$ legsta CR628336.1.gbk*
Found 'any2fasta' => /home/eric/miniconda3/bin/any2fasta
Found 'isPcr' => /home/eric/miniconda3/bin/isPcr
Loaded 2794 SBT types.
Loaded 39 flaA alleles
Loaded 58 pilE alleles
Loaded 77 asd alleles
Loaded 91 mip alleles
Loaded 105 mompS alleles
Loaded 58 proA alleles
Loaded 96 neuA alleles
FILE    SBT     flaA    pilE    asd     mip     mompS   proA    neuA
Running: any2fasta -q -u CR628336\.1\.gbk | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
CR628336.1.gbk  1       1       4       3       1       1       1       1
Running: any2fasta -q -u CR628336\.1\.gbk\.bz2 | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
CR628336.1.gbk.bz2      1       1       4       3       1       1       1       1
Running: any2fasta -q -u CR628336\.1\.gbk\.gz | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
CR628336.1.gbk.gz       1       1       4       3       1       1       1       1
Running: any2fasta -q -u CR628336\.1\.gbk\.zip | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
CR628336.1.gbk.zip      1       1       4       3       1       1       1       1
```

**Testing .fq, .fq.bz2, .fq.gz, .fq.zip:**
(250 bp simulated using ART)

```
(base) eric@ragnarok:~/Downloads$ legsta NC_018140_reads.fq*
Found 'any2fasta' => /home/eric/miniconda3/bin/any2fasta
Found 'isPcr' => /home/eric/miniconda3/bin/isPcr
Loaded 2794 SBT types.
Loaded 39 flaA alleles
Loaded 58 pilE alleles
Loaded 77 asd alleles
Loaded 91 mip alleles
Loaded 105 mompS alleles
Loaded 58 proA alleles
Loaded 96 neuA alleles
FILE    SBT     flaA    pilE    asd     mip     mompS   proA    neuA
Running: any2fasta -q -u NC_018140_reads\.fq | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_reads.fq      -       -       -       -       -       -       -       ?
Running: any2fasta -q -u NC_018140_reads\.fq\.bz2 | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_reads.fq.bz2  -       -       -       -       -       -       -       ?
Running: any2fasta -q -u NC_018140_reads\.fq\.gz | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_reads.fq.gz   -       -       -       -       -       -       -       ?
Running: any2fasta -q -u NC_018140_reads\.fq\.zip | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_reads.fq.zip  -       -       -       -       -       -       -       ?
```

The FASTQ tests are running, but the tool isn't finding anything, probably because the reads are too short. I'm not exactly sure where to find appropriate FASTQ-format files, so I just converted the FASTA-format `NC_018140` complete genome sequence to FASTQ by making the sequence one long read and then zipping it up in various ways:

**Testing .fq, .fq.bz2, .fq.gz, .fq.zip (1 long read):**

```
(base) eric@ragnarok:~/Downloads$ legsta NC_018140_1read.fq*
Found 'any2fasta' => /home/eric/miniconda3/bin/any2fasta
Found 'isPcr' => /home/eric/miniconda3/bin/isPcr
Loaded 2794 SBT types.
Loaded 39 flaA alleles
Loaded 58 pilE alleles
Loaded 77 asd alleles
Loaded 91 mip alleles
Loaded 105 mompS alleles
Loaded 58 proA alleles
Loaded 96 neuA alleles
FILE    SBT     flaA    pilE    asd     mip     mompS   proA    neuA
Running: any2fasta -q -u NC_018140_1read\.fq | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_1read.fq      734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140_1read\.fq\.bz2 | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_1read.fq.bz2  734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140_1read\.fq\.gz | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_1read.fq.gz   734     2       6       17      1       1       8       11
Running: any2fasta -q -u NC_018140_1read\.fq\.zip | isPcr stdin \/home\/eric\/miniconda3\/db\/ispcr\.tab stdout -out=fa -minPerfect=5 -tileSize=6 -maxSize=1200 -stepSize=5
NC_018140_1read.fq.zip  734     2       6       17      1       1       8       11
```

It appears to be uncompressing them with `any2fasta`.

@bgruening  Let me know if you want me to check any of the other [input formats](https://github.com/tseemann/any2fasta#motivation).

File: tools/legsta/legsta.xml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        #for $i in $contigs
+            '$i'
+        #end for
+
+        > '$output'
+    ]]></command>
+
+    <inputs>
+        <param type="data" name="contigs" format="genbank,embl,gff,gff3,fasta,fastq,clustal,stockholm" multiple="true" label="Contigs" help="One or more input files in FASTA, FASTQ, Genbank, EMBL, Clustal, Stockholm or GFF format." />
```
Comment (bgruening) [inline tools/legsta/legsta.xml:23]: If you investigation is correct, please add here fasta.gz, fastq.gz to the list.

File: tools/legsta/legsta.xml (hunk 2 above)
Comment (bgruening) [reply tools/legsta/legsta.xml:10]: Thanks.

**Comment by bgruening** [comment]
Ok, that looks good, the test needs to be fixed but than this looks pretty good.

**Comment by emarinier** [comment]
I ended up changing the Galaxy profile version from `20.05` to `20.01`, so that our institution's Galaxy build can use it without needing to immediately upgrade.

I downloaded a 20.01 version of Galaxy (`git clone -b release_20.01 https://github.com/galaxyproject/galaxy.git`) to test the change on my machine with `planemo s --galaxy_root [...]` and `planemo t --galaxy_root [...]`.

File: tools/legsta/legsta.xml (hunk 6 above)
Comment (emarinier) [reply tools/legsta/legsta.xml]: Added

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @emarinier!

**Pull request merged**